web/beacon-app/src/features/topics/components/TopicQueryResult/ViewingEvent.tsx

```diff
@@ -3,28 +3,28 @@ import { Heading } from '@rotational/beacon-core';
 
 import { CardListItem } from '@/components/common/CardListItem';
 
+import { getEventsPaginationCounter, getQueryPaginationCounter } from '../../utils';
 import MetaDataTable from './MetaDataTable';
-
 interface MetaDataProps {
   totalResults: number;
   totalEvents: string;
+  counter: number;
 }
 
-const ViewingEvent = ({ totalResults, totalEvents }: MetaDataProps) => {
+const ViewingEvent = ({ totalResults, totalEvents, counter }: MetaDataProps) => {
   return (
     <div className="mt-8">
       <Heading as="h2" className="mb-2 text-lg font-semibold">
         <Trans>
-          Query Results
-          <span className="ml-1 font-normal">
-            {' '}
-            - {String(totalResults ?? 'N/A')} of {totalEvents ?? 'N/A'}
+          Query Results -
+          <span className="ml-1 font-normal" data-testid="query-result-count">
+            {getQueryPaginationCounter(+totalResults, +totalEvents)}
           </span>
         </Trans>
       </Heading>
       <CardListItem className="!rounded-none">
-        <p>
-          <Trans>Viewing Event: 1 of 10</Trans>
+        <p data-testid="view-event">
+          <Trans>Viewing Event: {getEventsPaginationCounter(counter, +totalEvents)}</Trans>
```

Comment:
Should `totalResults` be used here instead of `totalEvents`? The total number of results should not be more than 10, for now. However, the total number of events could be more than 10.